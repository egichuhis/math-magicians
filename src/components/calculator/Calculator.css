.calculator-div {
  display: flex;
  justify-content: space-between;
  height: auto;
  padding-top: 40px;
}

.calculator {
  width: 420px;
}

.output {
  font-size: 24px;
  text-align: right;
  padding: 10px;
  background-color: #858693;
  color: #fff;
  border: 1px solid #ccc;
  border-radius: 3px;
}

.buttons {
  display: grid;
  grid-template-columns: repeat(4, 1fr);
}

.two-columns {
  grid-column: span 2;
}

button {
  font-size: 12px;
  padding: 16px;
  border: 1px solid #e0e0e0;
  cursor: pointer;
}

.ops-color {
  background-color: #f5913e;
}
