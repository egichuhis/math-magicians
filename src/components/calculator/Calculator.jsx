import React, { useState } from 'react';
import './Calculator.css';
import CalculatorOutput from './CalculatorOutput';
import CalculatorInputs from './CalculatorInputs';
import calculate from '../../logic/calculate';

const Calculator = () => {
  const [calculatorData, setCalculatorData] = useState({
    total: null,
    next: null,
    operation: null,
  });

  const handleClick = (buttonName) => {
    const newCalculatorData = calculate(calculatorData, buttonName);
    setCalculatorData(newCalculatorData);
  };

  return (
    <div className="calculator-div">
      <div>
        <h2>Let&apos;s do some math!</h2>
      </div>
      <div className="calculator">
        <CalculatorOutput calculatorData={calculatorData} />
        <CalculatorInputs handleClick={handleClick} />
      </div>

    </div>
  );
};

export default Calculator;
