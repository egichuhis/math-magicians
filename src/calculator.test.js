import React from 'react';
import { render } from '@testing-library/react';
import Calculator from './components/calculator/Calculator';

test('Calculator renders correctly', () => {
  const { container } = render(<Calculator />);
  expect(container).toMatchSnapshot();
});
