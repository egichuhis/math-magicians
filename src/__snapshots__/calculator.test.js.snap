// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator renders correctly 1`] = `
<div>
  <div
    class="calculator-div"
  >
    <div>
      <h2>
        Let's do some math!
      </h2>
    </div>
    <div
      class="calculator"
    >
      <div
        class="output"
        id="output"
      >
        0
      </div>
      <div
        class="buttons"
      >
        <button
          type="button"
        >
          AC
        </button>
        <button
          type="button"
        >
          +/-
        </button>
        <button
          type="button"
        >
          %
        </button>
        <button
          class="ops-color"
          type="button"
        >
          รท
        </button>
        <button
          type="button"
        >
          7
        </button>
        <button
          type="button"
        >
          8
        </button>
        <button
          type="button"
        >
          9
        </button>
        <button
          class="ops-color"
          type="button"
        >
          x
        </button>
        <button
          type="button"
        >
          4
        </button>
        <button
          type="button"
        >
          5
        </button>
        <button
          type="button"
        >
          6
        </button>
        <button
          class="ops-color"
          type="button"
        >
          -
        </button>
        <button
          type="button"
        >
          1
        </button>
        <button
          type="button"
        >
          2
        </button>
        <button
          type="button"
        >
          3
        </button>
        <button
          class="ops-color"
          type="button"
        >
          +
        </button>
        <button
          class="two-columns"
          type="button"
        >
          0
        </button>
        <button
          type="button"
        >
          .
        </button>
        <button
          class="ops-color"
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>
</div>
`;
